libs/state/selections/src/lib/accumulation-observable.ts

```diff
@@ -20,16 +20,36 @@ import {
   withLatestFrom,
 } from 'rxjs/operators';
 import { AccumulationFn, Accumulator } from './model';
+import { inject, InjectionToken } from '@angular/core';
 
 const defaultAccumulator: AccumulationFn = <T>(st: T, sl: Partial<T>): T => {
   return { ...st, ...sl };
 };
 
+/**
+ * Injection token for the default accumulator function.
+ *
+ * @example
+ * providers: [
+ *  {
+ *   provide: RX_ACCUMULATOR_FN,
+ *   useValue: (state, slice) => ({ ...state, ...slice })
+ *  }
+ * ]
+ */
+export const RX_ACCUMULATOR_FN = new InjectionToken<AccumulationFn>(
```

Comment:
shouldn't this be provided in root? How does it work if nothing is provided?

Reply:
If we provide it in root users cannot use different accumulator fn, I don't see a specific scenario where we have to specify multiple accumulators but I do think it can happen in large apps. What's your opinion?

If nothing is provided it uses the token factory which returns the default accumulator.

Reply:
Okay, if it anyway falls back to the default it's cool. But other question, is a non-root provided token non-treeshakable? 

Of course users have to be very careful where and when to provide a custom accumulator. Imo accumulators are very tightly couple to the state instance. See my comment here: https://github.com/rx-angular/rx-angular/pull/1649#discussion_r1405484501

Reply:
Non-root tokens are not tree-shakable as far as I know. Do we want to mark it provided in root? Knowing that we won't be able to specify multiple accumulators per application, which seems to go against your statement: 

> Imo accumulators are very tightly couple to the state instance

I would let users decide where they want to specify a custom accumulator and document that will be applied for children components as well.